lib/pages/new_group/widget/contacts_selection_list.dart

```diff
@@ -1,90 +1,100 @@
-import 'package:dartz/dartz.dart';
-import 'package:fluffychat/app_state/failure.dart';
-import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/domain/model/contact/contact_type.dart';
 import 'package:fluffychat/pages/new_group/widget/contacts_selection_list_style.dart';
 import 'package:fluffychat/presentation/model/presentation_contact.dart';
 import 'package:flutter/material.dart';
 
-import 'package:fluffychat/domain/app_state/contact/get_contacts_state.dart';
 import 'package:fluffychat/pages/new_group/selected_contacts_map_change_notifier.dart';
 import 'package:fluffychat/pages/new_private_chat/widget/expansion_contact_list_tile.dart';
-import 'package:fluffychat/pages/new_private_chat/widget/loading_contact_widget.dart';
 import 'package:fluffychat/pages/new_private_chat/widget/no_contacts_found.dart';
-import 'package:fluffychat/presentation/model/presentation_contact_success.dart';
+import 'package:matrix/matrix.dart';
 
 class ContactsSelectionList extends StatelessWidget {
   final SelectedContactsMapChangeNotifier selectedContactsMapNotifier;
-  final ValueNotifier<Either<Failure, Success>> contactsNotifier;
+  final ValueNotifier<List<PresentationContact>> presentationContactNotifier;
   final Function() onSelectedContact;
   final List<String> disabledContactIds;
+  final TextEditingController textEditingController;
+  final ValueNotifier<bool> isSearchModeNotifier;
 
   const ContactsSelectionList({
     Key? key,
-    required this.contactsNotifier,
+    required this.presentationContactNotifier,
     required this.selectedContactsMapNotifier,
     required this.onSelectedContact,
     this.disabledContactIds = const [],
+    required this.textEditingController,
+    required this.isSearchModeNotifier,
   }) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
     return ValueListenableBuilder(
-      valueListenable: contactsNotifier,
-      builder: (context, value, child) => value.fold(
-        (failure) => SliverToBoxAdapter(
-          child: Padding(
-            padding: ContactsSelectionListStyle.notFoundPadding,
-            child: NoContactsFound(
-              keyword: failure is GetContactsFailure ? failure.keyword : '',
-            ),
-          ),
-        ),
-        (success) {
-          if (success is PresentationExternalContactSuccess) {
-            return SliverToBoxAdapter(
-              child: _ContactItem(
-                selectedContactsMapNotifier: selectedContactsMapNotifier,
-                onSelectedContact: onSelectedContact,
-                contact: success.contact,
-                paddingTop: ContactsSelectionListStyle.listPaddingTop,
+      valueListenable: presentationContactNotifier,
+      builder: (context, presentationContact, child) {
+        final isSearchModeEnable =
+            textEditingController.text.isNotEmpty && isSearchModeNotifier.value;
+
+        final isExternalContact = textEditingController.text.isValidMatrixId &&
```

Comment:
you should move this logic into search function
if keyword is external contact, return list with 1 external contact